R/utils.R

```diff
@@ -348,3 +348,30 @@ is_tab_active_js <- function(id, name) {
     id, name
   )
 }
+
+#' Check if an argument is a bounded numeric vector of length 3
+#' Must follow form `c(value, min, max)`
+#' @noRd
+check_range_slider <- function(
+    value, lower = -Inf, upper = Inf, finite = TRUE, null.ok = FALSE) { # nolint: object_name.
+  is_numeric <- checkmate::test_numeric(
+    value,
+    len = 3, any.missing = FALSE, finite = finite, null.ok = null.ok
+  )
+  is_bounded <- is_numeric && checkmate::test_number(
+    value[[1]],
+    lower = max(lower, value[[2]]),
+    upper = min(upper, value[[3]]),
+    null.ok = null.ok,
+    finite = finite
+  )
+  if (isFALSE(is_numeric) || isFALSE(is_bounded)) {
```

Comment:
Technically we could only test if `is_bounded` is FALSE, but I think this adds readability.

`is_numeric && checkmate::test_number(...)` is being used a few lines before so that we can safely do `value[[2]]` and `value[[3]]`